src/components/Authentication/SignUp.js

```diff
@@ -115,19 +109,24 @@ export default function SignUp() {
         setConfirmPassword({...confirmPassword, password: e.target.value});
         if (e.target.value === userData.password) {
             setConfirmPassword( { ...confirmPassword, isEqualToPassword: true});
+            if (registrationError.message === "Both passwords should match") {
+                setRegistrationError({ ...registrationError, message: "", hasError: false})
+            }
         }
         else {
+            setRegistrationError({ ...registrationError, message: "Both passwords should match", hasError: true})
             setConfirmPassword( { ...confirmPassword, isEqualToPassword: false});
         }
     }
 
 
-    const PageError =  registrationError.hasError ? (
+    const PageError =  React.useMemo(() => (registrationError.message !== ""
+        ? (
         <Typography align="center" color="#DA3A16">
             {registrationError.message}
         </Typography>
-    ) :
-        (<React.Fragment/>);
+        )
+        : null), [registrationError]);
 
     const ProgramSelect = (
```

Comment:
Can this be created as a shared component? It exists in both SignUp.js and EditProfile.js. This is the only duplicate in this code, so should be a quick fix.

Reply:
yes,lost of page implement this error thing. We can create a custom one so its displays the errorMEssage for every page

Reply:
I'm referring to ProgramSelect const, not the error. The same const exists in 2 different files, so maybe we can put it in a separate file as a common resource to reduce duplicates.